pkg/extensions/tbtc/recovery/electrs.go

```diff
@@ -0,0 +1,77 @@
+package recovery
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+)
+
+type httpClient interface {
+	Post(url string, contentType string, body io.Reader) (*http.Response, error)
+	Get(url string) (*http.Response, error)
+}
+
+// ElectrsConnection exposes a native API for interacting with an electrs http API.
+type ElectrsConnection struct {
+	apiURL string
+	client httpClient
+}
+
+// NewElectrsConnection is a constructor for ElectrsConnection.
+func NewElectrsConnection(apiURL string) *ElectrsConnection {
+	return &ElectrsConnection{
+		apiURL: apiURL,
+		client: http.DefaultClient,
+	}
+}
+
+func (e *ElectrsConnection) setClient(client httpClient) {
+	e.client = client
+}
+
+// Broadcast broadcasts a transaction the configured bitcoin network.
+func (e ElectrsConnection) Broadcast(transaction string) error {
+	if e.apiURL == "" {
+		for i := 0; i < 5; i++ {
+			logger.Warningf("Please broadcast Bitcoin transaction %s", transaction)
+		}
+		return nil
+	}
+	resp, err := e.client.Post(fmt.Sprintf("%s/tx", e.apiURL), "text/plain", strings.NewReader(transaction))
+	if err != nil {
+		return err
+	}
+	if resp.StatusCode != 200 {
+		return fmt.Errorf("something went wrong with broadcast: [%s], transaction: [%s]", resp.Status, transaction)
+	}
+	logger.Info("successfully broadcast the bitcoin transaction")
```

Comment:
Does the `resp` contain any transaction ID we could log here as well?

Reply:
Yup! we're now reading the response body and including it in the logs:

```
2021-04-27T17:13:44.478-0400	INFO	bitcoin	successfully broadcast the bitcoin transaction: 4d8cd9e11af8a421a5519bbc5172dfb17c408557973c3998077e12bd0d74561c
```
![image](https://user-images.githubusercontent.com/1045160/116314355-fac61180-a77c-11eb-99e3-6f08555b0f0b.png)

![image](https://user-images.githubusercontent.com/1045160/116314377-04e81000-a77d-11eb-8d73-21b18006377c.png)